pegasus/forms/hoc_signup_2014.rb

```diff
@@ -9,23 +9,47 @@ def self.normalize(data)
-    result[:organization_name_s] = required stripped data[:organization_name_s]
-    result[:event_type_s] = enum(data[:event_type_s].to_s.strip.downcase, ['in_school', 'out_of_school'])
-    result[:event_location_s] = stripped data[:event_location_s]
+    result[:organization_name_s] = stripped data[:organization_name_s]
+    result[:event_type_s] = required enum(data[:event_type_s].to_s.strip.downcase, ['in_school', 'out_of_school'])
     result[:entire_school_flag_b] = stripped data[:entire_school_flag_b]
     result[:send_posters_flag_b] = stripped data[:send_posters_flag_b]
     result[:send_posters_address_s] = stripped data[:send_posters_address_s]
     result[:special_event_flag_b] = stripped data[:special_event_flag_b]
     result[:special_event_details_s] = stripped data[:special_event_details_s]
     result[:match_volunteer_flag_b] = stripped data[:match_volunteer_flag_b]
-    result[:hoc_country_s] = enum(data[:hoc_country_s].to_s.strip.downcase, HOC_COUNTRIES.keys)
+    result[:hoc_country_s] = required enum(data[:hoc_country_s].to_s.strip.downcase, HOC_COUNTRIES.keys)
     result[:hoc_company_s] = nil_if_empty data[:hoc_company_s]
 
-    #If the user's school is not found via the school dropdown
-    result[:school_name_s] = stripped data[:school_name_s]
-    result[:school_city_s] = stripped data[:school_city_s]
-    result[:school_state_s] = stripped data[:school_state_s]
-    result[:school_zip_s] = stripped data[:school_zip_s]
+    # If the user's school is not found via the school dropdown
+    # we still need to require location for US in school events to match
+    # to NCES id for census data.
+    if result[:event_type_s] == 'in_school' && result[:hoc_country_s] == 'US'
+      result[:event_location_s] = required stripped data[:event_location_s]
+    else
+      result[:event_location_s] = stripped data[:event_location_s]
+    end
+
+    #If the user has an in-school US event, they will fill out 2017 census questions.
+    result[:role_s] = data[:role_s]
+    result[:hoc_s] = data[:hoc_s]
+    result[:after_school_s] = data[:after_school_s]
+    result[:ten_hours_s] = data[:ten_hours_s]
+    result[:twenty_hours_s] = data[:twenty_hours_s]
+    result[:other_cs_b] = data[:other_cs_b]
+    result[:followup_frequency_s] = data[:followup_frequency_s]
+    result[:followup_more_s] = stripped data[:followup_more_s]
+    result[:topic_blocks_b] = data[:topic_blocks_b]
+    result[:topic_text_b] = data[:topic_text_b]
+    result[:topic_robots_b] = data[:topic_robots_b]
+    result[:topic_internet_b] = data[:topic_internet_b]
+    result[:topic_security_b] = data[:topic_security_b]
+    result[:topic_data_b] = data[:topic_data_b]
+    result[:topic_web_design_b] = data[:topic_web_design_b]
+    result[:topic_game_design_b] = data[:topic_game_design_b]
+    result[:topic_other_b] = data[:topic_other_b]
+    result[:topic_dont_know_b] = data[:topic_dont_know_b]
+    result[:pledge_b] = data[:pledge_b]
+
```

Comment:
Why is this editing `HocSignup2014`?

Shouldn't it be [HocSignup2017](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/forms/hoc_signup_2017.rb)?